### DISCUSSION TITLE
Hangfire integration - Application failed to start

### DISCUSSION THREAD

**Discussion by jerinantony11** [abpframework/abp#16064] [Q&A] [answered]
Hi,
I have an existing ABP application(Version 6) which is up and running. I'm trying to implement the Hangfire worker as per this documentation [https://docs.abp.io/en/abp/6.0/Background-Workers-Hangfire].  I have followed all the steps mentioned but the application failed start with below error message. I think the documentation is incomplete. There may be a few more steps to complete this. Can some one help me with this please.

Volo.Abp.AbpInitializationException: An error occurred during the initialize Volo.Abp.Modularity.OnPreApplicationInitializationModuleLifecycleContributor phase of the module Volo.Abp.BackgroundWorkers.Hangfire.AbpBackgroundWorkersHangfireModule, Volo.Abp.BackgroundWorkers.Hangfire, Version=6.0.3.0, Culture=neutral, PublicKeyToken=null: An exception was thrown while activating ?:Volo.Abp.Hangfire.AbpHangfireBackgroundJobServer -> ?:Hangfire.JobStorage.. See the inner exception for details.
 ---> Autofac.Core.DependencyResolutionException: An exception was thrown while activating ?:Volo.Abp.Hangfire.AbpHangfireBackgroundJobServer -> ?:Hangfire.JobStorage.
 ---> System.InvalidOperationException: JobStorage.Current property value has not been initialized. You must set it before using Hangfire Client or Server API.
   at Hangfire.JobStorage.get_Current()

**Comment by maliming** [comment, marked as answer]
hi

https://github.com/abpframework/abp/issues/11069

**Reply by jerinantony11** [reply]
Thanks. this one helped. I also had to install a nuget package [Hangfire.MemoryStorage]  to get this working.
I do think the documentation need updating. because the application just breaks and fails to run after following the documentation.

**Reply by maliming** [reply]
sure
